crates/ra_ide/src/lib.rs

```diff
@@ -470,6 +470,13 @@ impl Analysis {
         position: FilePosition,
         new_name: &str,
     ) -> Cancelable<Option<RangeInfo<SourceChange>>> {
+        let tokens = tokenize(new_name);
+        if tokens.len() != 1
+            || (tokens[0].kind != SyntaxKind::IDENT && tokens[0].kind != SyntaxKind::UNDERSCORE)
+        {
+            return Ok(None);
+        }
```

Comment:
Let's push this ever further down, to `references::rename`. That way, we can write a test!

Reply:
The tests work fine before the move, but you're right in that `references::rename` is the better place for it.